### PR TITLE
testutil/compose: refactor auto

### PR DESCRIPTION
**Pull request by corverroos**
Refactors the `auto` command by moving it into the main `compose` repo. This simplifies the smoke tests and cleans up the compose cli `main` package.

category: test
ticket: #992

### CODE REVIEW

File: testutil/compose/auto.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+type AutoConfig struct {
+	// Dir is the directory to use for compose artifacts.
+	Dir string
+	// AlertTimeout is the timeout to collect alerts before shutdown. Zero disables timeout.
+	AlertTimeout time.Duration
+	// SudoPerms enables changing all compose artefacts file permissions using sudo.
+	SudoPerms bool
+	// Print generated docker-compose.yml files.
+	PrintYML bool
+	// RunTmplFunc allows arbitrary overrides if the run step template.
```
Comment (xenowits) [inline testutil/compose/auto.go:39]:
```suggestion
	// RunTmplFunc allows arbitrary overrides in the run step template.
```

File: testutil/compose/auto.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+	RunTmplFunc func(*TmplData)
+	// DefineTmplFunc allows arbitrary overrides if the define step template.
```
Comment (xenowits) [inline testutil/compose/auto.go:41]:
```suggestion
	// DefineTmplFunc allows arbitrary overrides in the define step template.
```

File: testutil/compose/auto.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+		if step.RunStep { // Continue below if final run step.
+			break
+		}
+
+		if err := execUp(ctx, conf.Dir); err != nil {
+			return err
+		}
+	}
+
+	if conf.AlertTimeout != 0 {
```
Comment (xenowits) [inline testutil/compose/auto.go:108]: nit:
```suggestion
	if conf.AlertTimeout > 0 {
```

**Review by xenowits — APPROVED** [review]

**Review by dB2510 — APPROVED** [review]

File: testutil/compose/auto.go [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+	}
+
+	return nil
+}
+
+// execUp executes `docker-compose up`.
+func execUp(ctx context.Context, dir string) error {
+	// Build first so containers start at the same time below.
+	log.Info(ctx, "Executing docker-compose build")
+	cmd := exec.CommandContext(ctx, "docker-compose", "build", "--parallel")
```
[the remark this answers is not in the document]
Comment (corverroos) [reply testutil/compose/auto.go:204]: noted thanks

**Pull request merged**